core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -40,50 +40,48 @@ module Streamly.Internal.Data.MutArray.Type
 
     -- ** Casting
     , cast
-    , castUnsafe
+    , castUnsafe -- XXX unsafeCast
     , asBytes
-    , asPtrUnsafe
-    , asUnpinnedPtrUnsafe
+    , asPtrUnsafe -- XXX rename to unsafePinnedAsPtr?
+    , asUnpinnedPtrUnsafe -- XXX rename to unsafeAsPtr
 
     -- ** Construction
-    , nil
+    , empty
 
     -- *** New
     -- | New arrays are always empty arrays with some reserve capacity to
     -- extend the length without reallocating.
-    , new
-    , newArrayWith
-    , pinnedNew
-    , pinnedNewBytes
-    , pinnedNewAligned
+    , emptyOf
+    , newArrayWith -- emptyAlignedWith
+    , pinnedEmptyOf
+    , pinnedNewAligned -- XXX not required
+    -- , new -- uninitialized array of specified length
 
     -- *** Cloning
     , clone
     , pinnedClone
 
     -- *** Slicing
     -- | Get a subarray without copying
-    , getSliceUnsafe
+    , getSliceUnsafe -- XXX unsafeGetSlice
```

Comment:
We can use `Unsafe` as a suffix over a prefix.
`Unsafe` is auxiliary information, not the primary.